### PR TITLE
ENH: Adding new package impact-t.

### PR DESCRIPTION
**Pull request by hhslepicka**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by hhslepicka** [comment]
@ChristopherMayes could you please confirm agreeing to be a Maintainer?

**Comment by ChristopherMayes** [comment]
I agree to be a maintainer.

**Comment by hhslepicka** [comment]
@conda-forge-admin, please lint

**Comment by hhslepicka** [comment]
@conda-forge/staged-recipes this is ready for a review/merge.

Thank you!

**Comment by hhslepicka** [comment]
@conda-forge-admin, please lint

**Comment by hhslepicka** [comment]
@conda-forge/staged-recipes I have no idea why it is complaining about overlinking.
I would appreciate if someone could have a look.

File: recipes/impact-t/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+requirements:
+  build:
+    - make
+    - {{ compiler('c') }}                    # [not win]
+    - {{ compiler('m2w64_c') }}              # [win]
+    - {{ compiler('fortran') }}              # [not win]
+    - {{ compiler('m2w64_fortran') }}        # [win]
+    - cmake
+    - {{ mpi }}  # [mpi != 'nompi']
+  host:
```
Comment (chrisburr) [inline recipes/impact-t/meta.yaml:53]: This definitely needs to be changed:
```suggestion
  host:
    - {{ mpi }}  # [mpi != 'nompi']
```

I'm not very familiar with mpi so there might be other things as well, it's worth checking if anything else needs changing to match the example in the docs: https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=mpich#message-passing-interface-mpi

File: recipes/impact-t/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,79 @@
+{% set name = "impact-t" %}
+{% set version = "2.0" %}
+
+{% set build = 1000 %}
```
Comment (chrisburr) [inline recipes/impact-t/meta.yaml:4]: This should start from 0:
```suggestion
{% set build = 0 %}
```

File: recipes/impact-t/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    - {{ compiler('m2w64_fortran') }}        # [win]
+    - cmake
+    - {{ mpi }}  # [mpi != 'nompi']
+  host:
+  run:
+    - {{ mpi }}  # [mpi != 'nompi']
+
+test:
+  commands:
+    - which ImpactTexe  # [unix]
```
Comment (chrisburr) [inline recipes/impact-t/meta.yaml:59]: Can you test these commands at all, even if it's just running `--help` or `--version`?

Also the Windows build currently isn't tested at all so something should be added for that as well (`ImpactTexe --help` would cover everything if that's an option)

File: recipes/impact-t/meta.yaml (hunk 1 above)
Comment (hhslepicka) [reply recipes/impact-t/meta.yaml]: Addressed via https://github.com/conda-forge/staged-recipes/pull/13701/commits/b3f2e1ea43915c3324f073bf35f25910e2d9bf0b

File: recipes/impact-t/meta.yaml (hunk 2 above)
Comment (hhslepicka) [reply recipes/impact-t/meta.yaml]: Addressed via https://github.com/conda-forge/staged-recipes/pull/13701/commits/49c05e7cf6304bd40867832a6f5fe339a461a6fa

I followed the pattern at some recipes in which the number start at 1000 to favor the non-mpi build.
I switched it back to 0.

File: recipes/impact-t/meta.yaml (hunk 3 above)
Comment (hhslepicka) [reply recipes/impact-t/meta.yaml:58]: So here is the tricky part about it... `ImpactTexe` has no --help or any other thing... it just runs when you call and it doesn't take parameters.
I would need to check for existance of the binary.

Comment (hhslepicka) [reply recipes/impact-t/meta.yaml:58]: Addressed via https://github.com/conda-forge/staged-recipes/pull/13701/commits/a455233fb859e8b38e07c0ce83afa477b5f6ee29

**Comment by hhslepicka** [comment]
@chrisburr this one is good to go... thank you for the help and review. 👍

File: recipes/impact-t/meta.yaml (hunk 2 above)
Comment (chrisburr) [reply recipes/impact-t/meta.yaml]: Yeah that's a legacy thing: https://conda-forge.org/docs/maintainer/maintainer_faq.html#mfaq-build-number-1000

I've opened a PR to update the docs: https://github.com/conda-forge/conda-forge.github.io/pull/1246

**Pull request merged**